model/course.py

```diff
@@ -105,16 +105,16 @@ def modify_course(TAs, student_nicknames):
                   methods=['GET', 'POST', 'PUT', 'DELETE'])
 @login_required
 def grading(user, course_name, student):
-    course = Course(course_name).obj
+    course = Course(course_name)
+
     if not course:
         return HTTPError('Course not found.', 404)
-    permission = perm(course, user)
-    if not permission:
+    if not course.permission(user, Course.Permission.VIEW):
         return HTTPError('You are not in this course.', 403)
-    if student not in course.student_nicknames.keys():
+    if student not in course.obj.student_nicknames.keys():
```

Comment:
We don't need to access `obj` here. `MongoBase` impls `__getattr__` and `__setattr__` to make it can almost be treated as underlying `mongoengine.Document`.

p.s. Don't forget to update other lines include `.obj` in this file. (except line 73)

Reply:
https://github.com/Normal-OJ/Back-End/pull/188/commits/d7fcfb381cb93d03ef9b91114f8c82da02fb07c0